orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/ConfigStageInjectionTransform.java

```diff
@@ -75,17 +72,107 @@ private void replaceStages(PipelineTemplate pipelineTemplate) {
   }
 
   private void injectStages(PipelineTemplate pipelineTemplate) {
-    List<StageDefinition> configStages = templateConfiguration.getStages()
+    // Create initial graph via dependsOn.
+    createDag(pipelineTemplate.getStages());
+
+    // Handle stage injections.
+    injectStages(pipelineTemplate.getStages(), pipelineTemplate.getStages());
+    injectStages(templateConfiguration.getStages(), pipelineTemplate.getStages());
+  }
+
+  private static void createDag(List<StageDefinition> stages) {
+    stages
       .stream()
+      .filter(s -> !s.getDependsOn().isEmpty() && s.getInject() == null)
+      .forEach(s -> s.setRequisiteStageRefIds(
+        stages
+          .stream()
+          .filter(parentCandidate -> s.getDependsOn().contains(parentCandidate.getId()))
+          .map(StageDefinition::getId)
+          .collect(Collectors.toList())
+      ));
+  }
+
+  private static void injectStages(List<StageDefinition> stages, List<StageDefinition> templateStages) {
```

Comment:
They're just in-class helper methods. I tend to like to break methods up where I can so it's easier to follow each piece. Since they're only useful as far as the internals of this class, I don't think it makes a lot of sense to have a helper class.